e2e/mock/peer.go

```diff
@@ -76,3 +79,11 @@ func NewPeer(TxC chan struct{}, credentials credentials.TransportCredentials) (*
 
 	return instance, nil
 }
+
+func (p *Peer) Suspend() {
+	p.Add(1)
```

Comment:
first of all, there's race in accessing `wg`, from Go doc for `sync.WaitGroup`
> // Note that calls with a positive delta that occur when the counter is zero
> // must happen before a Wait.

secondly, the `p.Wait()` at line 36 blocks the process, which may block the sender of `TxC` when its buffer is full.

lastly, current implementation requires user to call `Suspend` only **before** server actually starts, which limits its use case. Although this is fine for now